realm/src/androidTest/java/io/realm/RealmLinkTests.java

```diff
@@ -463,4 +463,53 @@ public void testSubquery() {
         RealmResults<Owner> subOwners = owners.where().equalTo("cat.name", "Blackie").findAll();
         assertEquals(1, subOwners.size());
     }
+
+    public void testIsNull() {
+        RealmResults<Owner> owners1 = testRealm.where(Owner.class).isNull("dogs").findAll();
+        assertEquals(0, owners1.size());
+
+        RealmResults<Owner> owners2 = testRealm.where(Owner.class).isNull("cat").findAll();
+        assertEquals(0, owners2.size());
+
+        // remove cat and dogs and query again
+        testRealm.beginTransaction();
+        testRealm.clear(Dog.class);
+        testRealm.clear(Cat.class);
+        testRealm.commitTransaction();
+
+        RealmResults<Owner> owners3 = testRealm.where(Owner.class).isNull("dogs").findAll();
+        assertEquals(1, owners3.size());
+
+        RealmResults<Owner> owners4 = testRealm.where(Owner.class).isNull("cat").findAll();
+        assertEquals(1, owners4.size());
+    }
+
+    public void testIsNotNull() {
```

Comment:
These test test both LINK and LINK_LIST. It would be much more readable and easier to see if functionality is tested by splitting it into: testRelationshipIsNotNull() and testRealmListIsNotNull() and similar for isNull()